src/python/pants/backend/python/macros/pex_binaries_from_sources_caof.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os.path
+from typing import Iterable
+
+from pants.backend.python.macros.caof_utils import OVERRIDES_TYPE, flatten_overrides
+from pants.engine.target import InvalidFieldException
+
+
+class PexBinariesFromSourcesCAOF:
```

Comment:
I couldn't find a good definition of what CAOF really means (what makes it "context-aware").

So when in rome...

Reply:
It just means a context aware object factory. Here you're context aware (constructed with a parse context) and you're an object factory (you create targets with that context).

Reply:
The parse context is only used for creating the targets and for error reporting. We can probably live without the error reporting bit if needed. I didn't see a way to create targets in the other macros without the parse context. If it exists, happy to try it out!

Reply:
Exactly. For old style macros to do anything useful it's needed. Thus CAOF.

This word probably sprang as a differentiator from the fact you could register "targets", "objects", and CAOFs in Pants V1. A CAOF has the constructor you use here and ~must use the context to do anything useful.

An object was simply a registered alias for a plain old python type that got exposed in BUILD files for direct construction. An example was:
```python
python_library(
  ...
  provides=setup_py(...)
)
```
There any `python_library` could gain the ability to define what today is a separate `python_distribution` target. To do so you used the `setup_py` object.